File: gravitee-apim-console-webui/src/entities/management-api-v2/api/v4/mcp.ts
/*
 * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export class MCP {
  enabled: boolean;
  tools: MCPTool[];
}

export class MCPTool {
  name?: string;
  description?: string;
  inputSchema?: MCPToolInputSchema;
}

export class MCPToolInputSchema {
  type: string;
  properties?: MCPToolProperties;
  required?: string[];
}

export class MCPToolProperties {
  [property: string]: unknown;
}

File: gravitee-apim-definition/gravitee-apim-definition-model/src/main/java/io/gravitee/definition/model/v4/mcp/MCP.java
/*
 * Copyright © 2015 The Gravitee team (http://gravitee.io)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.gravitee.definition.model.v4.mcp;

import com.fasterxml.jackson.annotation.JsonIgnore;
import io.gravitee.definition.model.Plugin;
import io.swagger.v3.oas.annotations.media.Schema;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor
@Builder(toBuilder = true)
@Data
@Schema(name = "MCPV4")
public class MCP implements Serializable {

    @Builder.Default
    protected boolean enabled = true;

    @Builder.Default
    private List<Tool> tools = new ArrayList<>();
    // TODO MCP: Return with correct information for plugins?
    //    @JsonIgnore
    //    public List<Plugin> getPlugins() {
    //        return List.of(new Plugin("server", "mcp-server"));
    //    }
}
